src/Components/PostsList/EditPost/index.jsx

```diff
@@ -0,0 +1,178 @@
+import React from "react";
+import { withStyles } from "@material-ui/core/styles";
+import styles from "./styles";
+import {
+  TextField,
+  Grid,
+  Card,
+  CardHeader,
+  CardActions,
+  CardContent,
+  Button,
+  Tooltip,
+  IconButton
+} from "@material-ui/core";
+import DeleteIcon from "@material-ui/icons/Delete";
+import PhotoCameraIcon from "@material-ui/icons/PhotoCamera";
+import SaveIcon from "@material-ui/icons/Save";
+import { withRouter } from "react-router-dom";
+
+class EditPost extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleTitleChange = this.handleTitleChange.bind(this);
+    this.handleTextChange = this.handleTextChange.bind(this);
+    this.handlePhotoChange = this.handlePhotoChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.state = { post: { title: "", text: "" }, selectedFile: null };
+  }
+  handleTitleChange(event) {
+    this.setState({ post: { ...this.state.post, title: event.target.value } });
```

Comment:
ta i poniższa metoda są dość podobne, możę by je jakoś zmergować, np

handleChange(event,field){
this.setState({ post: { ...this.state.post, [field]: event.target.value } });
}